File: LATIHAN1DPBO2023/PYTHON/Crud.py
class Crud:
      
    # membuat method Update 
    def ubah(nim, masukan):
        name = []
        nimubah = []
        ps = []
        fakultas = []
        
        i=0;
        for masukan in masukan:
            if (nim == masukan.get_nim()):
                # print(str(i+1) + ".", masukan.get_name(), "-", masukan.get_nim(), "-", masukan.get_ps(), "-", masukan.get_fakultas())
                name = str(input())
                nimubah = str(input())
                ps = str(input())
                fakultas = str(input())
                
                masukan.set_name(name)
                masukan.set_nim(nimubah)
                masukan.set_ps(ps)
                masukan.set_fakultas(fakultas)
                i+1
    
    def hapus(nim, masukan):
        i = 0;
        for data in masukan:
            if (nim == data.get_nim()):
                del masukan[i]
                i+= 1

File: LATIHAN1DPBO2023/JAVA/Crud.java
import java.util.ArrayList;
import java.util.Scanner;

public class Crud {
    /* konstruktor tanpa parameter */
    public Crud()
    {
    
    }
    /* method crud */
    public void ubah(String nim, ArrayList<Mhs> list)
    {
        Scanner sc = new Scanner(System.in);

        // /* deklrasi variabel nama, nim yang diubah, program studi, dan fakultas*/
        String name;
        String nimubah;
        String ps;
        String fakultas;

        int i = 0;
        /* perulangan */
        for(i = 0; i < list.size(); i++)
        {
            if(nim.equals(list.get(i).getNim()))
            {
                System.out.println("Berhasil menemukan NIM, silahkan masukan data yang ingin diubah!");
                name = sc.next();
                nimubah = sc.next();
                ps = sc.next();
                fakultas = sc.next();
    

                list.get(i).setName(name);
                list.get(i).setNim(nimubah);
                list.get(i).setPs(ps);
                list.get(i).setFakultas(fakultas);
            }
        }
    }
    public void hapus(String nim, ArrayList<Mhs> list)
    {
        int i;
        for(i = 0; i < list.size(); i++)
        {
            if(nim.equals(list.get(i).getNim()))
            {
                list.remove(list.get(i));
            }
        }
    }
    public void mashok(ArrayList<Mhs> list, Mhs temp)
    {
        list.add(temp);
    }
    public void ingpomaseh(ArrayList<Mhs> list)
    {
        int i = 0;
        for(i = 0; i < list.size(); i++)
        {
        	System.out.println(Integer.toString(i + 1) + ". " + list.get(i).getName() + " " + list.get(i).getNim() + " " + list.get(i).getPs() + " "+list.get(i).getFakultas());
        }

    }
}
